src/topology/schema.rs

```diff
@@ -30,82 +31,59 @@ use crate::{
 ///
 /// Finally, The merged definition (named `Definition 1 & 2`), and `Definition 4` are merged
 /// together to produce the new `Definition` returned by this method.
-pub(super) fn merged_definition(
+pub fn merged_definition(
     inputs: &[OutputId],
     config: &dyn ComponentContainer,
-    cache: &mut HashMap<Vec<OutputId>, Definition>,
+    cache: &mut HashMap<(bool, Vec<OutputId>), Definition>,
 ) -> Definition {
     // Try to get the definition from the cache.
-    if let Some(definition) = cache.get(inputs) {
+    if let Some(definition) = cache.get(&(config.schema_enabled(), inputs.to_vec())) {
         return definition.clone();
     }
 
-    // This should be "never", but that isn't supported yet (coming soon!)
-    let mut definition: Option<Definition> = None;
+    // TODO: special case when inputs is empty?
```

Comment:
Anything to do here before merging? Remove comment, or create an issue, or solve the TODO?

Reply:
this was fixed